mcf-browser/src/GraphExplorer.tsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {Component} from 'react';
+
+import {LoadingSpinner} from './LoadingSpinner';
+import {PageBar} from './PageBar';
+
+const NODES_PER_PAGE = 25;
+
+ interface GraphExplorerPropType {
+   /**
+    * Indicates if uploaded files are currently being parsed.
+    */
+   loading: boolean;
+
+   /**
+    * IDs for nodes stored in App's state which are the subject nodes of
+    * triples from any parsed files.
+    */
+   subjNodes: string[];
+
+   /**
+    * Set id parameter in url to the given id. Used when user clicks a
+    * subject node to explore.
+    */
+   goToId: Function;
```

Comment:
nit: usually define the type for a function like `(dcid: string) => void` 
nit: maybe `onNodeClick` or something

Reply:
done!